ee/server/service/devices.go

```diff
@@ -35,9 +41,12 @@ func (svc *Service) TriggerMigrateMDMDevice(ctx context.Context, host *fleet.Hos
 		bre.InternalErr = ctxerr.New(ctx, "macOS migration webhook URL not configured")
 	case !host.IsOsqueryEnrolled(), !host.MDMInfo.IsDEPCapable(), !host.MDMInfo.IsEnrolledInThirdPartyMDM():
 		bre.InternalErr = ctxerr.New(ctx, "host not eligible for macOS migration")
+	case host.RefetchCriticalQueriesUntil != nil && host.RefetchCriticalQueriesUntil.After(svc.clock.Now()):
+		// the webhook has already been triggered successfully recently (within the
+		// refetch critical queries delay), so do as if it did send it successfully
+		// but do not re-send.
+		return nil
```

Comment:
Would like to hear your thoughts here, about returning 204 vs some error code if a request is made too soon after a successful one? My reasoning was that I don't think there's much use to return an error code to the frontend and worry the user with an error in this scenario, and if we'd want to notify them that it was an unnecessary request we'd need more UI designs (to show a helpful explanation message and not "Something's gone wrong" in this case) for little benefit, I think?

Reply:
fully agree 👍